azavea/tilejson.io
# Fix share link

## Overview

The share link doesn't work because the `index.html` being shared has an `ol` error. To fix this, this PR does the following:
 - [x] Adds an edit mode flag to the app state
 - [x] Allows a user to just view the TileJSON from a Gist by visiting the URI `/g/:gistID/view`
 - [x] Removes the `index.html` as one of the files shared to the Gist
 - [x] Updates the link on a successful share from the bl.ocks link to the new `tilejson.io/g/:gistID/view` link

Connects [#162100057](https://www.pivotaltracker.com/story/show/162100057)

## Testing Instructions

 * Serve locally or test on Netlify PR site. 
 * Add a couple (more than 1) layers to map. Here are some tile sets: https://a.tiles.azavea.com/climate/tmean_05/{z}/{x}/{y}.png, https://a.tiles.azavea.com/climate/tmean_06/{z}/{x}/{y}.png, https://a.tiles.azavea.com/climate/tmean_07/{z}/{x}/{y}.png
 * Login via GitHub
 * Share the map, providing options you desire. Remember that there are *7 options*: `title` (the map title), `description` (the map description), `shareGist` (whether to share a link to the Gist), `shareTileJSONLink` (whether to share a link to the site), `shareBase` (whether to include a base layer when sharing), `shareDiff` (whether to include a diff option in the shared link), and `defaultToDiff` (whether to default to the diff mode). Whatever you pick now, you can change manually on the Gist you create to test the view link. 
 * You will get a link that looks like `tilejson.io/g/:id/view`. Copy the ID. You can open your gist at `gist.github.com/:id` and a view at `deploy-preview-97--tilejson.netlify.com/g/:id/view`.
 * Ensure that settings you configured for your share page work correctly. Edit your Gist to invoke different settings and make sure those, too, work correctly. Below is a list of expected behavior based on values provided to each of the *7 options*.

`title` should appear in the card on the right corner of the map. If `description` is not an empty string, it should appear as an expandable portion of the same card. The Gist button, Diff button, and TileJSON.io link/branding on the top left should only be visible if `shareGist`, `shareDiff`, and `shareTileJSONLink` are `true` respectively. The base layer should be blank if `shareBase` is false and should be the current base layer when share was clicked if it is true. Lastly, `defaultToDiff` if true, should make the default view of this view page the diff mode. 

<img width="1341" alt="screen shot 2018-11-27 at 9 35 36 am" src="https://user-images.githubusercontent.com/3959096/49088706-d79f7c80-f227-11e8-9af3-841a8b45dc6a.png">
